File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_change(int change);

int main(void)
{
    // declare change variable
    int change_due;
    do
    {
        // assign the change variable
        change_due = get_int("Change due, please: ");
    }
    // validate the change variable
    while (change_due < 1 );

    // find minimum # of coins to give back as change
    // Call calculate_change function
    int coins_back = calculate_change(change_due);
    // Final Step print result
    printf("%i\n", coins_back);
}

// Subtract function
int calculate_change(int change)
{
    int remaining_change;
    // calculate # of quarters (highest coin value) to give back and subtract it from the change
    int quarters = (change / 25);
    remaining_change = change - (quarters * 25);
    // calculate # of dimes (next highest coin value) to return and subtract from remaining change
    int dimes = remaining_change / 10;
    remaining_change = remaining_change - (dimes * 10);
    // calculate # of nickels (next highest coin value) to return and subtract from remaining change
    int nickels = remaining_change / 5;
    remaining_change = remaining_change - (nickels * 5);
    // calculate # of pennies (next highest coin value) to return and subtract from remaining change
    int pennies = remaining_change / 1;
    remaining_change = remaining_change - (pennies * 1);

    // find the sum total of all the coins needed
    int total_coin_number = quarters + dimes + nickels + pennies;

    // return the total # of coins
    return total_coin_number;
}

File: sentimental-cash/cash.py
from cs50 import get_float

# CASH revisited:
# define a prompt function


def get_flt(prompt):
    while True:
        # test for a non-numeric answer
        try:
            change_due = get_float(prompt)
            # test for positive integer
            if change_due > 0:
                return change_due
            else:
                print("Please enter a positive dollar amount between .01 and 100")
        except ValueError:
            pass


def calculate_change(change):
    # convert change to cents -avoid flt division
    change_to_cents = round(change * 100)
    # use integer division //
    quarters = (change_to_cents // 25)
    # capture remainder with modulo
    remaining_change = change_to_cents % 25
    dimes = remaining_change // 10
    remaining_change = remaining_change % 10
    nickels = remaining_change // 5
    remaining_change = remaining_change % 5
    pennies = remaining_change // 1
    remaining_change = remaining_change % 1

    # find sum total of all the coins needed
    # print("Quarters:", quarters, "Dimes:", dimes, "Nickels:", nickels, "Pennies: ", pennies)
    total_coin_number = quarters + dimes + nickels + pennies
    return total_coin_number

# defining main function


def main():
    # prompt user for their change back
    n = get_flt("Change: ")
    # find min # of change to return
    coins_back = calculate_change(n)
    print(coins_back)


# Execute the main program
main()
